### DISCUSSION TITLE
Avoid removing and re-adding integration?

### DISCUSSION THREAD

**Discussion by canton7** [nathanmarlor/foxess_modbus#51] [General]
Hi,

Is there any way to avoid having to remove and re-add the integration when you make config changes? It means that HA loses the history for all of the sensors. It's the one thing that's keeping me using foxess-ha still -- I don't want to reset all of my lovely graphs whenever there's an udpate.

Thanks!

**Comment by nathanmarlor** [comment]
@canton7 it'd just mean more work to make everything backwards compatible, when re-adding the component is a 30s job. You shouldn't lose any history as the sensor entity IDs are static and generates static unique IDs. I've updated a couple of times today and still have history?

![Screenshot_20230316-215136.png](https://user-images.githubusercontent.com/3436911/225760475-809f3975-0572-4e1a-9acf-aacd459e11de.png)

**Reply by canton7** [reply]
Hmm, I'll take another look tomorrow. Last time I removed/re-added, I lost the sensor history, but maybe that was due to them gaining a state_class.

**Reply by canton7** [reply]
I stand corrected! They stuck around that time.

Although the breakage this time was a bit insidious -- the symptom was everything just being unavailable, and the instructions to remove and re-add were buried several release notes down, so you had to go clicking to expand them all.

**Reply by nathanmarlor** [reply]
Agreed - the only resolution I can think off is to check all config options are available at init and throw an exception if not; this will bubble up to the devices page as 'failed to load' with an error in the logs. Probably a better approach if/when breaking changes are needed in the future.

**Reply by canton7** [reply]
Would it be worth putting a version number into the configuration?

I've found that to be very helpful in the past -- you can immediately see whether the config is old and how old it is, which means you can either run a series of database-style migrations to bring it up to the latest version, or abort and complain.
